libcontainer/configs/validate/validator.go

```diff
@@ -383,7 +383,7 @@ func scheduler(config *configs.Config) error {
 	if s.Policy == "" {
 		return errors.New("scheduler policy is required")
 	}
-	if s.Nice < -20 || s.Nice > 19 {
+	if (s.Policy == specs.SchedOther || s.Policy == specs.SchedBatch) && s.Nice < -20 || s.Nice > 19 {
 		return fmt.Errorf("invalid scheduler.nice: %d", s.Nice)
```

Comment:
Not "nice"

Reply:
can you describe it in detail?

Reply:
The error message says this is `scheduler.nice`, but it is actually `schedule.policy`

Reply:
Also, the `if` expression should be split to two statements. One for nice, one for policy.